File: src/main/java/com/github/svegon/capi/util/FogContext.java
package com.github.svegon.capi.util;

import com.google.common.base.Preconditions;
import net.minecraft.client.render.BackgroundRenderer;
import net.minecraft.client.render.FogShape;

public final class FogContext {
    private final BackgroundRenderer.FogData fogData;

    public FogContext(BackgroundRenderer.FogData fogData) {
        this.fogData = fogData;
    }

    public BackgroundRenderer.FogType fogType() {
        return fogData.fogType;
    }

    public float fogStart() {
        return fogData.fogStart;
    }

    public void fogStart(float start) {
        fogData.fogStart = start;
    }

    public float fogEnd() {
        return fogData.fogEnd;
    }

    public void fogEnd(float end) {
        fogData.fogEnd = end;
    }

    public FogShape fogShape() {
        return fogData.fogShape;
    }

    public void fogShape(FogShape shape) {
        fogData.fogShape = Preconditions.checkNotNull(shape);
    }
}

File: src/main/kotlin/io/github/svegon/mclientapi/util/FogContext.kt
package io.github.svegon.mclientapi.util

import net.minecraft.client.render.BackgroundRenderer
import net.minecraft.client.render.FogShape

class FogContext(private val fogData: BackgroundRenderer.FogData) {
    val fogType: BackgroundRenderer.FogType
        get() {
            return fogData.fogType
        }

    val fogStart: Float
        get() {
            return fogData.fogStart
        }

    val fogEnd: Float
        get() {
            return fogData.fogEnd
        }

    var fogShape: FogShape
        get() {
            return fogData.fogShape
        }
        set(shape) {
            fogData.fogShape = shape
        }
}
